src/Core/Components/Tabs/FluentTabs.razor.cs

```diff
@@ -154,41 +154,44 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
             _jsModuleOverflow = await JSRuntime.InvokeAsync<IJSObjectReference>("import", JAVASCRIPT_FILE.FormatCollocatedUrl(LibraryConfiguration));
 
             var horizontal = Orientation == Orientation.Horizontal;
-            await _jsModuleOverflow.InvokeVoidAsync("fluentOverflowInitialize", _dotNetHelper, Id, horizontal, FLUENT_TAB_TAG,25);
+            await _jsModuleOverflow.InvokeVoidAsync("fluentOverflowInitialize", _dotNetHelper, Id, horizontal, FLUENT_TAB_TAG, 25);
         }
     }
 
     private async Task HandleOnTabChangedAsync(TabChangeEventArgs args)
     {
         var tabId = args?.ActiveId;
-        if (tabId is not null && _tabs.TryGetValue(tabId, out FluentTab? tab))
+        var tab = _tabs.FirstOrDefault(i => i.Id == tabId);
+
+        if (tab is not null && _tabs.Contains(tab))
         {
             await OnTabChange.InvokeAsync(tab);
-            ActiveTabId = tabId;
+            ActiveTabId = tabId!;
             await ActiveTabIdChanged.InvokeAsync(tabId);
         }
     }
 
     internal int RegisterTab(FluentTab tab)
     {
-        _ = _tabs.TryAdd(tab.Id!, tab);
+        _tabs.Add(tab);
         return _tabs.Count - 1;
     }
 
-    internal async Task UnregisterTabAsync(FluentTab tab)
+    internal async Task UnregisterTabAsync(string id)
```

Comment:
Why did you change this signature ? Using the Tab object could be more upgradable. And the Tab ID can be set to `null`.

Reply:
> Blazor is recycling HashCodes so when a FluentTab is removed, the next tab is assigned the same HashCode (and memory pointer?) and the UnregisterTabAsync method starts acting on the wrong tab.

Blazor itself does not recycle hash codes.

In my opinion, the problem probably comes from somewhere else, where the dictionary reorganization is not adapted to this component.

I understand the advantage of using an ID that avoids passing the wrong object reference. On the other hand, it may trigger another problem, as `Id` is nullable, so it's possible to have no identifier.

Reply:
Id is indeed nullable but in our use cases it can't be null as it is assigned a value in the FluentTab constructor.

Reply:
Even if the constructor assigns a default value to ID, nothing prevents the developer from defining `Id=“@null”` :-)

Reply:
> Even if the constructor assigns a default value to ID, nothing prevents the developer from defining `Id=“@null”` :-)

That is why I said 'in our use cases' 😉 Developers will always get themselves in trouble if we let them.

Reply:
As discussed internally. We'll go for the quick fix now. We can revisit when we work on the v5 component